predictive_test_selection/collect_test_failure_data.py

```diff
@@ -0,0 +1,168 @@
+
+
+class CommonFileExtension(IntFlag):
+    other = auto()
+    mo = auto()
+    po = auto()
+    pyc = auto()
+    json = auto()
+    dat = auto()
+    pyi = auto()
+    py = auto()
+
+
+class DataParser:
+    """
+        This class fetches all information regarding a specific code change and the tests that are run on this specific commit
+        and sends it all to the influxdb database.
+
+        Information regarding the code change:
+            x commit_hash
+            x n_changes_3  \
+            x n_changes_14  -> Changes made to the files involved in this change in the last 3, 14 and 56 days
+            x n_changes_56 /
+            x file_cardinality -> Number of files involved in this change
+            x file_extension -> File extensions of the files involved in this change
+
+        Information regarding the tests:
+            x test_fqn -> fully qualified name of the test case: <test_file_name>.<test_case>
+            x test_failed -> integer denoting whether this test failed or not.
+    """
+
+    def __init__(self):
+        self.code_change_data: Mapping[str, Any] = {}
+        self.test_result_data: Mapping[str, int] = {}
+
+    def parse_code_change(self) -> None:
+        """
+        Collect all data pertaining to the code change in the currently checked out branch:
+        - the commit hash
+        - the number of modified files
+        - the file extensions of modified files
+        - the number of changes to the modified files in the past 3, 14 and 56 days
+        """
+        data = {
+            "commit_hash": None,
+            "file_cardinality": None,
+            "file_extension": None,
+            "n_changes": None,
+        }
+
+        # Get latest commit hash:
+        data["commit_hash"] = subprocess.check_output(["git", "log", "--pretty=%H", "-1"]).strip()
+
+        # Get current branch
+        cmd = ["git", "rev-parse", "--abbrev-ref", "HEAD"]
+        current_branch = subprocess.check_output(cmd).strip().decode()
+
+        cmd = ["git", "diff", f"master...{current_branch}", "--name-only"]
+        changed_files = [line for line in subprocess.check_output(cmd, env=os.environ.copy()).decode().split("\n") if line]
+
+        data["file_cardinality"] = len(changed_files)
+
+        def get_extension_vector() -> List[int]:
+            extensions = set([os.path.splitext(file)[1].replace(".", "") for file in changed_files])
+
+            out = [0] * 8
+            known_exts = [ext for ext in CommonFileExtension]
+            for ext in extensions:
+                try:
+                    idx = known_exts.index(CommonFileExtension[ext])
+                except KeyError:
+                    idx = known_exts.index(CommonFileExtension["other"])
+                out[idx] = 1
+            return out
+
+        data["file_extension"] = get_extension_vector()
+
+        def get_changes() -> List[int]:
+            n_changes: List[int] = []
+            for n_days_ago in [3, 14, 56]:
```

Comment:
The list `[3, 14, 56]` can be considered a configuration parameter of this script. As such, It would make it into a global variable to make that clear.